--- net_main.c
+++ net_main.c
@@ -49,8 +49,8 @@
 static double	slistStartTime;
 static int		slistLastShown;
 
-static void Slist_Send(void);
-static void Slist_Poll(void);
+static void Slist_Send(void *unused);
+static void Slist_Poll(void *unused);
 PollProcedure	slistSendProcedure = {NULL, 0.0, Slist_Send};
 PollProcedure	slistPollProcedure = {NULL, 0.0, Slist_Poll};
 
@@ -312,7 +312,7 @@
 }
 
 
-static void Slist_Send(void)
+static void Slist_Send(void *unused)
 {
 	for (net_driverlevel=0; net_driverlevel < net_numdrivers; net_driverlevel++)
 	{
@@ -328,7 +328,7 @@
 }
 
 
-static void Slist_Poll(void)
+static void Slist_Poll(void *unused)
 {
 	for (net_driverlevel=0; net_driverlevel < net_numdrivers; net_driverlevel++)
 	{
