--- net_udp.c
+++ net_udp.c
@@ -44,6 +44,7 @@
 
 static int net_acceptsocket = -1;		// socket for fielding new connections
 static int net_controlsocket;
+static struct in_addr	net_hostaddr;
 static int net_broadcastsocket = 0;
 static struct qsockaddr broadcastaddr;
 
@@ -59,6 +60,23 @@
 	char	buff[MAXHOSTNAMELEN];
 	struct qsockaddr addr;
 	char *colon;
+	int		i;
+
+	// Check for -ip command line parameter
+	net_hostaddr.s_addr = INADDR_ANY;
+	i = COM_CheckParm ("-ip");
+	if (i)
+	{
+		if (i < com_argc-1)
+		{
+			net_hostaddr.s_addr = inet_addr(com_argv[i+1]);
+			if (net_hostaddr.s_addr == INADDR_NONE)
+				Sys_Error ("%s is not a valid IP address", com_argv[i+1]);
+			Con_Printf("Binding to IP address %s\n", com_argv[i+1]);
+		}
+		else
+			Sys_Error ("-ip requires an IP address");
+	}
 	
 	if (COM_CheckParm ("-noudp"))
 		return -1;
@@ -138,7 +156,7 @@
 		goto ErrorReturn;
 
 	address.sin_family = AF_INET;
-	address.sin_addr.s_addr = INADDR_ANY;
+	address.sin_addr = net_hostaddr;
 	address.sin_port = htons(port);
 	if( bind (newsocket, (void *)&address, sizeof(address)) == -1)
 		goto ErrorReturn;
