--- cl_main.c
+++ cl_main.c
@@ -20,6 +20,7 @@
 // cl_main.c  -- client main loop
 
 #include "quakedef.h"
+#include "cl_replay.h"
 
 // we need to declare some mouse variables here, because the menu system
 // references them even when on a unix system.
@@ -126,6 +127,11 @@
 
 	cls.demoplayback = cls.timedemo = false;
 	cls.signon = 0;
+
+#ifdef __EMSCRIPTEN__
+	COM_SwitchGame(NULL);
+	CL_Replay_Reset();
+#endif
 }
 
 void CL_Disconnect_f (void)
@@ -753,5 +759,9 @@
 	Cmd_AddCommand ("stop", CL_Stop_f);
 	Cmd_AddCommand ("playdemo", CL_PlayDemo_f);
 	Cmd_AddCommand ("timedemo", CL_TimeDemo_f);
+
+#ifdef __EMSCRIPTEN__
+	CL_Replay_Init ();
+#endif
 }
 
