--- host.c
+++ host.c
@@ -44,6 +44,7 @@
 int			host_framecount;
 
 int			host_hunklevel;
+int			fs_hunklevel;
 
 int			minimum_memory;
 
@@ -479,8 +480,13 @@
 	Con_DPrintf ("Clearing memory\n");
 	D_FlushCaches ();
 	Mod_ClearAll ();
-	if (host_hunklevel)
-		Hunk_FreeToLowMark (host_hunklevel);
+	{
+		int lowmark = host_hunklevel;
+		if (fs_hunklevel > lowmark)
+			lowmark = fs_hunklevel;
+		if (lowmark)
+			Hunk_FreeToLowMark (lowmark);
+	}
 
 	cls.signon = 0;
 	memset (&sv, 0, sizeof(sv));
