--- net_dgrm.c
+++ net_dgrm.c
@@ -54,6 +54,9 @@
 
 #include "quakedef.h"
 #include "net_dgrm.h"
+#ifdef __EMSCRIPTEN__
+#include <emscripten/emscripten.h>
+#endif
 
 // these two macros are to make the code more readable
 #define sfunc	net_landrivers[sock->landriver]
@@ -516,11 +521,12 @@
 static int		testDriver;
 static int		testSocket;
 
-static void Test_Poll(void);
+static void Test_Poll(void *unused);
 PollProcedure	testPollProcedure = {NULL, 0.0, Test_Poll};
 
-static void Test_Poll(void)
+static void Test_Poll(void *unused)
 {
+	(void)unused;
 	struct qsockaddr clientaddr;
 	int		control;
 	int		len;
@@ -644,11 +650,12 @@
 static int		test2Driver;
 static int		test2Socket;
 
-static void Test2_Poll(void);
+static void Test2_Poll(void *unused);
 PollProcedure	test2PollProcedure = {NULL, 0.0, Test2_Poll};
 
-static void Test2_Poll(void)
+static void Test2_Poll(void *unused)
 {
+	(void)unused;
 	struct qsockaddr clientaddr;
 	int		control;
 	int		len;
@@ -1150,6 +1150,9 @@
 		if (MSG_ReadByte() != CCREP_SERVER_INFO)
 			continue;
 
+		NET_SlistParseAggregatedList(net_landriverlevel);
+		continue;
+
 		dfunc.GetAddrFromName(MSG_ReadString(), &readaddr);
 		// search the cache for this server
 		for (n = 0; n < hostCacheCount; n++)
@@ -1302,6 +1313,10 @@
 					continue;
 				}
 			}
+#ifdef __EMSCRIPTEN__
+			if (ret == 0)
+				emscripten_sleep(1);
+#endif
 		}
 		while (ret == 0 && (SetNetTime() - start_time) < 2.5);
 		if (ret)
